toolchain/sem_ir/typed_insts.h

```diff
@@ -21,6 +21,8 @@
 //   value. This includes instructions that produce an abstract value, such as a
 //   `Namespace`, for which a placeholder type should be used.
 // - Up to two `[...]Id` members describing the contents of the struct.
```

Comment:
Maybe we should say: up to two members of types derived from `IndexBase` rather than inaccurately saying `` `[...]Id` ``.

Reply:
Though even that's not quite right. `IntId` and `BuiltinInstKind` are the exceptions. The real rule is that they need to be listed in the `IdKind` enum and implement the `FromRaw`/`ToRaw` protocol in `inst.h`.

Reply:
Ah but one is type_id, okay nvm. This one threw me off but that was because I forgot the type_id rule.

Reply:
A couple more observations though:
- `AnyFoundationDecl` and `AnyBranch` have an `int32_t` member, rather than something like an `ElementIndex`. `ToRaw(int32_t)` does not appear to be defined.
- While `IntId` is, `BuiltinInstKind` is _not_ in the `IdKind` enum. But the latter is also not a field in any typed inst. Rather than having a `BuiltinInst` typed inst, we now have a separate struct for each one, introduced in be5db6e1cdf6a815f3.
- The idea that builtins don't have a location id comes from the old text around `SemIR::Builtin`, a type that no longer exists. I can't find a reason why this would be true in the code, the parsing adds a TokenIndex at the location of a builtin instruction that should come with a source location. I am guessing this was maybe referring to the _definition_ of builtins? And that this just no longer exists, as they are proper builtin instructions now? As such I think I would remove that clause and just say all instructions come with a location?